nixos/doc/manual/development/settings-options.xml

```diff
@@ -167,13 +167,51 @@ in {
 
     # We know that the `user` attribute exists because we set a default value
     # for it above, allowing us to use it without worries here
-    users.users.${cfg.settings.user} = {}
+    users.users.${cfg.settings.user} = {};
 
     # ...
   };
 }
 </programlisting>
    </example>
+   <section xml:id="sec-settings-attrs-options">
+    <title>Option declarations for attributes</title>
+    <para>
+     It can be very valuable to declare options for a subset of <literal>settings</literal> attributes in order to check their type and show their documentation in the manual. This can be done using <xref linkend='sec-freeform-modules'/>.
```

Comment:
```suggestion
     Some <literal>settings</literal> attributes may deserve some extra care. They may need a different type, default or merging behavior, or they are essential options that should show their documentation in the manual. This can be done using <xref linkend='sec-freeform-modules'/>.
```